### ISSUE TITLE
Plugin to merge frame to video with audio

### ISSUE THREAD

**Issue by ruah1984** [deepfakes/faceswap#63]
HI ,
can we add a plugin to merge frame to video with audio?

**Comment by Clorr** [comment]
Hello,

I don't understand the request. Can you provide more info please ?

**Comment by Apollo122** [comment]
He asks a plugin which converts frames into the video and adds original audio from source video into fake video.

**Comment by Clorr** [comment]
:-D Frame, not flame!

I currently have in mind the @shaoanlu method with moviepy, but I have not much time to work on it. There is a draft pull request if you want to have a look

**Comment by Apollo122** [comment]
In my view this can be archived with FFMPEG easily. So it's not a priority. I'm waiting for GAN with masking it will be a game changer if its like in those gifs

**Comment by Clorr** [comment]
Can you provide some guidelines to @ruah1984 , if he needs some ?

**Comment by Clorr** [comment]
For the gan plugin, the classes are somewhat ready in https://github.com/shaoanlu/faceswap-GAN/blob/master/temp/faceswap_GAN_keras.ipynb I have to merge them, hopefully later today

**Comment by ruah1984** [comment]
sorry for the typo.
no hurry i think i can wait until GAN plugin ready in deepfakes/faceswap. 
thanks.

**Comment by Apollo122** [comment]
sure.  i use these commands.

frames to video
ffmpeg -f image2 -framerate <fps_of_source_video> -i "img (%d).jpg" -vcodec libx264 -profile:v high444 -refs 16 -crf 0 -preset ultrafast input_0.mp4

adding sound
ffmpeg -i input_0.mp4 -i source.mp4 -c copy -map 0:0 -map 1:1 -shortest out.mp4

before using these commands, you should copy source frames to merged folder and say NO when it asks if it should overwrite existing files. You should do thing step because there might be some frames without faces in them or algorithm may have failed to detect faces for specific frames. If you fail to do this step correctly final video and audio won't sync.

**Comment by Clorr** [comment]
I'm closing this as I consider it to be handled in #10
